lib/_tls_wrap.js

```diff
@@ -614,18 +609,12 @@ TLSSocket.prototype._releaseControl = function() {
 };
 
 TLSSocket.prototype._finishInit = function() {
-  // `newSession` callback wasn't called yet
-  if (this._newSessionPending) {
-    this._securePending = true;
-    return;
-  }
```

Comment:
are we sure about removing this check? This function might be called outside of this module context.

Reply:
See one of the comments above regarding this.

Reply:
This does not answer my question. This function might be called from outside this module.

Reply:
Well, I mean, I suppose someone could reimplement the whole logic around `_newSessionPending` in user-land. But right now that `_newSessionPending` is set within `onnewsession` which only runs on the server and so it was moved into that branch instead. The reason this was added was to support async sessions, as far as I can tell and this is the code in question:

```js
function onnewsession(key, session) {
   [removed]

    owner._newSessionPending = false;
    if (owner._securePending)
      owner._finishInit();
    owner._securePending = false;
  };


  owner._newSessionPending = true;
  if (!owner.server.emit('newSession', key, session, done))
    done();
}